compile failed when deserilize msg struct

Reproduction steps:

1. enable serde in cargo.toml
```toml
[dependencies]
serde = "1.0"
serde_derive = "1.0"
serde_json = "1.0"
rclrs = "0.3"
rosidl_runtime_rs = "0.3"
std_msgs = "*"
nav_msgs = { version = "*", features = ["serde"] }
```
2. serde json
```rust
let odom = nav_msgs::msg::Odometry::default();
let json_str = serde_json::to_string(&odom).unwrap();
```
3. `colcon build`
4. compile error info
```log
--- stderr: fk_dumper_node                            
error[E0277]: the trait bound `[f64; 36]: Deserialize<'_>` is not satisfied
    --> /d/_work_projects/code_ros2/rust_ws/install/geometry_msgs/share/geometry_msgs/rust/src/msg.rs:156:21
     |
156  |     pub covariance: [f64; 36],
     |                     ^^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `[f64; 36]`
     |
     = help: the following other types implement trait `Deserialize<'de>`:
               &'a [u8]
               [T; 0]
               [T; 10]
               [T; 11]
               [T; 12]
               [T; 13]
               [T; 14]
               [T; 15]
             and 26 others
note: required by a bound in `next_element`
    --> /home/fawdlstty/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.183/src/de/mod.rs:1733:12
     |
1731 |     fn next_element<T>(&mut self) -> Result<Option<T>, Self::Error>
     |        ------------ required by a bound in this associated function
1732 |     where
1733 |         T: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `SeqAccess::next_element`

error[E0277]: the trait bound `[f64; 36]: Deserialize<'_>` is not satisfied
    --> /d/_work_projects/code_ros2/rust_ws/install/geometry_msgs/share/geometry_msgs/rust/src/msg.rs:156:21
     |
156  |     pub covariance: [f64; 36],
     |                     ^^^^^^^^^ the trait `Deserialize<'_>` is not implemented for `[f64; 36]`
     |
     = help: the following other types implement trait `Deserialize<'de>`:
               &'a [u8]
               [T; 0]
               [T; 10]
               [T; 11]
               [T; 12]
               [T; 13]
               [T; 14]
               [T; 15]
             and 26 others
note: required by a bound in `next_value`
    --> /home/fawdlstty/.cargo/registry/src/index.crates.io-6f17d22bba15001f/serde-1.0.183/src/de/mod.rs:1872:12
     |
1870 |     fn next_value<V>(&mut self) -> Result<V, Self::Error>
     |        ---------- required by a bound in this associated function
1871 |     where
1872 |         V: Deserialize<'de>,
     |            ^^^^^^^^^^^^^^^^ required by this bound in `MapAccess::next_value`
```
6. generated file in `/d/_work_projects/code_ros2/rust_ws/install/geometry_msgs/share/geometry_msgs/rust/src/msg.rs:156:21`
```rust
// Corresponds to geometry_msgs__msg__AccelWithCovariance
#[repr(C)]
#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
#[derive(Clone, Debug, PartialEq, PartialOrd)]
pub struct AccelWithCovariance {
    pub accel: crate::msg::rmw::Accel,
    pub covariance: [f64; 36],
}
```

--- a/rosidl_generator_rs/resource/Cargo.toml.em
+++ b/rosidl_generator_rs/resource/Cargo.toml.em
@@ -6,6 +6,7 @@ edition = "2021"
 [dependencies]
 rosidl_runtime_rs = "0.3"
 serde = { version = "1", optional = true, features = ["derive"] }
+serde-big-array = "0.5.1"
 @[for dep in dependency_packages]@
 @(dep) = "*"
 @[end for]@

--- a/rosidl_generator_rs/resource/Cargo.toml.em
+++ b/rosidl_generator_rs/resource/Cargo.toml.em
@@ -6,14 +6,14 @@ edition = "2021"
 [dependencies]
 rosidl_runtime_rs = "0.3"
 serde = { version = "1", optional = true, features = ["derive"] }
-serde-big-array = "0.5.1"
+serde-big-array = { version = "0.5.1", optional = true }
 @[for dep in dependency_packages]@
 @(dep) = "*"
 @[end for]@
 
 [features]
 @{
-serde_features = ["dep:serde", "rosidl_runtime_rs/serde"]
+serde_features = ["dep:serde", "dep:serde-big-array", "rosidl_runtime_rs/serde"]
 for dep in dependency_packages:
 	serde_features.append("{}/serde".format(dep))
 }@

--- a/rosidl_generator_rs/resource/msg.rs.em
+++ b/rosidl_generator_rs/resource/msg.rs.em
@@ -5,6 +5,7 @@ TEMPLATE(
     package_name=package_name, interface_path=interface_path,
     msg_specs=msg_specs,
     get_rs_name=get_rs_name, get_rmw_rs_type=get_rmw_rs_type,
+    pre_field_serde=pre_field_serde,
     get_idiomatic_rs_type=get_idiomatic_rs_type,
     constant_value_to_rs=constant_value_to_rs)
 }@
@@ -16,6 +17,7 @@ TEMPLATE(
     package_name=package_name, interface_path=interface_path,
     msg_specs=msg_specs,
     get_rs_name=get_rs_name, get_rmw_rs_type=get_rmw_rs_type,
+    pre_field_serde=pre_field_serde,
     get_idiomatic_rs_type=get_idiomatic_rs_type,
     constant_value_to_rs=constant_value_to_rs)
 }@

--- a/rosidl_generator_rs/resource/msg_idiomatic.rs.em
+++ b/rosidl_generator_rs/resource/msg_idiomatic.rs.em
@@ -26,7 +26,7 @@ type_name = msg_spec.structure.namespaced_type.name
 #[derive(Clone, Debug, PartialEq, PartialOrd)]
 pub struct @(type_name) {
 @[for member in msg_spec.structure.members]@
-    pub @(get_rs_name(member.name)): @(get_idiomatic_rs_type(member.type)),
+    @(pre_field_serde(member.type))pub @(get_rs_name(member.name)): @(get_idiomatic_rs_type(member.type)),
 @[end for]@
 }
 

--- a/rosidl_generator_rs/resource/msg_rmw.rs.em
+++ b/rosidl_generator_rs/resource/msg_rmw.rs.em
@@ -38,7 +38,7 @@ extern "C" {
 #[derive(Clone, Debug, PartialEq, PartialOrd)]
 pub struct @(type_name) {
 @[for member in msg_spec.structure.members]@
-    pub @(get_rs_name(member.name)): @(get_rmw_rs_type(member.type)),
+    @(pre_field_serde(member.type))pub @(get_rs_name(member.name)): @(get_rmw_rs_type(member.type)),
 @[end for]@
 }
 

--- a/rosidl_generator_rs/resource/srv.rs.em
+++ b/rosidl_generator_rs/resource/srv.rs.em
@@ -12,6 +12,7 @@ TEMPLATE(
     package_name=package_name, interface_path=interface_path,
     msg_specs=req_res_specs,
     get_rs_name=get_rs_name, get_rmw_rs_type=get_rmw_rs_type,
+    pre_field_serde=pre_field_serde,
     get_idiomatic_rs_type=get_idiomatic_rs_type,
     constant_value_to_rs=constant_value_to_rs)
 }@
@@ -49,6 +50,7 @@ TEMPLATE(
     package_name=package_name, interface_path=interface_path,
     msg_specs=req_res_specs,
     get_rs_name=get_rs_name, get_rmw_rs_type=get_rmw_rs_type,
+    pre_field_serde=pre_field_serde,
     get_idiomatic_rs_type=get_idiomatic_rs_type,
     constant_value_to_rs=constant_value_to_rs)
 }@

--- a/rosidl_generator_rs/rosidl_generator_rs/__init__.py
+++ b/rosidl_generator_rs/rosidl_generator_rs/__init__.py
@@ -101,6 +101,7 @@ def generate_rs(generator_arguments_file, typesupport_impls):
             'Services template file %s not found' % template_file
 
     data = {
+        'pre_field_serde': pre_field_serde,
         'get_rmw_rs_type': make_get_rmw_rs_type(args['package_name']),
         'get_rs_name': get_rs_name,
         'get_idiomatic_rs_type': make_get_idiomatic_rs_type(args['package_name']),
@@ -274,6 +275,14 @@ def constant_value_to_rs(type_, value):
 #     - BoundedSequence
 #     - UnboundedSequence
 
+
+def pre_field_serde(type_):
+    if isinstance(type_, Array) and type_.size > 32:
+        return '#[cfg_attr(feature = "serde", serde(with = "serde_big_array::BigArray"))]\n    '
+    else:
+        return ''
+
+
 def make_get_idiomatic_rs_type(package_name):
     get_rmw_rs_type = make_get_rmw_rs_type(package_name)
     def get_idiomatic_rs_type(type_):
@@ -335,7 +344,9 @@ def make_get_rmw_rs_type(package_name):
         elif isinstance(type_, UnboundedSequence):
             return 'rosidl_runtime_rs::Sequence<{}>'.format(get_rmw_rs_type(type_.value_type))
         elif isinstance(type_, BoundedSequence):
-            return 'rosidl_runtime_rs::BoundedSequence<{}, {}>'.format(get_rmw_rs_type(type_.value_type), type_.maximum_size)
+            return 'rosidl_runtime_rs::BoundedSequence<{}, {}>'.format(
+                get_rmw_rs_type(type_.value_type),
+                type_.maximum_size)
 
         assert False, "unknown type '%s'" % type_.typename
     return get_rmw_rs_type

--- a/rosidl_generator_rs/rosidl_generator_rs/__init__.py
+++ b/rosidl_generator_rs/rosidl_generator_rs/__init__.py
@@ -174,7 +174,6 @@ def generate_rs(generator_arguments_file, typesupport_impls):
 
     return 0
 
-
 def get_rs_name(name):
     keywords = [
         # strict keywords
@@ -189,7 +188,6 @@ def get_rs_name(name):
     ]
     # If the field name is a reserved keyword in Rust append an underscore
     return name if not name in keywords else name + '_'
-
 
 def escape_string(s):
     s = s.replace('\\', '\\\\')
@@ -258,7 +256,7 @@ def constant_value_to_rs(type_, value):
     assert False, "unknown constant type '%s'" % type_
 
 # Type hierarchy:
-#
+# 
 # AbstractType
 # - AbstractNestableType
 #   - AbstractGenericString
@@ -300,7 +298,6 @@ def make_get_idiomatic_rs_type(package_name):
         else:
             return get_rmw_rs_type(type_)
     return get_idiomatic_rs_type
-
 
 def make_get_rmw_rs_type(package_name):
     def get_rmw_rs_type(type_):

--- a/rosidl_generator_rs/rosidl_generator_rs/__init__.py
+++ b/rosidl_generator_rs/rosidl_generator_rs/__init__.py
@@ -285,7 +285,6 @@ def pre_field_serde(type_):
 
 def make_get_idiomatic_rs_type(package_name):
     get_rmw_rs_type = make_get_rmw_rs_type(package_name)
-
     def get_idiomatic_rs_type(type_):
         if isinstance(type_, UnboundedString) or isinstance(type_, UnboundedWString):
             return 'std::string::String'

--- a/rosidl_generator_rs/rosidl_generator_rs/__init__.py
+++ b/rosidl_generator_rs/rosidl_generator_rs/__init__.py
@@ -174,6 +174,7 @@ def generate_rs(generator_arguments_file, typesupport_impls):
 
     return 0
 
+
 def get_rs_name(name):
     keywords = [
         # strict keywords
@@ -188,6 +189,7 @@ def get_rs_name(name):
     ]
     # If the field name is a reserved keyword in Rust append an underscore
     return name if not name in keywords else name + '_'
+
 
 def escape_string(s):
     s = s.replace('\\', '\\\\')
@@ -256,7 +258,7 @@ def constant_value_to_rs(type_, value):
     assert False, "unknown constant type '%s'" % type_
 
 # Type hierarchy:
-# 
+#
 # AbstractType
 # - AbstractNestableType
 #   - AbstractGenericString
@@ -285,6 +287,7 @@ def pre_field_serde(type_):
 
 def make_get_idiomatic_rs_type(package_name):
     get_rmw_rs_type = make_get_rmw_rs_type(package_name)
+
     def get_idiomatic_rs_type(type_):
         if isinstance(type_, UnboundedString) or isinstance(type_, UnboundedWString):
             return 'std::string::String'
@@ -297,6 +300,7 @@ def make_get_idiomatic_rs_type(package_name):
         else:
             return get_rmw_rs_type(type_)
     return get_idiomatic_rs_type
+
 
 def make_get_rmw_rs_type(package_name):
     def get_rmw_rs_type(type_):
@@ -344,9 +348,7 @@ def make_get_rmw_rs_type(package_name):
         elif isinstance(type_, UnboundedSequence):
             return 'rosidl_runtime_rs::Sequence<{}>'.format(get_rmw_rs_type(type_.value_type))
         elif isinstance(type_, BoundedSequence):
-            return 'rosidl_runtime_rs::BoundedSequence<{}, {}>'.format(
-                get_rmw_rs_type(type_.value_type),
-                type_.maximum_size)
+            return 'rosidl_runtime_rs::BoundedSequence<{}, {}>'.format(get_rmw_rs_type(type_.value_type), type_.maximum_size)
 
         assert False, "unknown type '%s'" % type_.typename
     return get_rmw_rs_type

